internal/oci/uvm.go

```diff
@@ -313,6 +314,49 @@ func parseAnnotationsString(a map[string]string, key string, def string) string
 	return def
 }
 
+// handleAnnotationKernelDirectBoot handles parsing annotationKernelDirectBoot and setting
+// implied annotations from the result.
+func handleAnnotationKernelDirectBoot(ctx context.Context, a map[string]string, lopts *uvm.OptionsLCOW) {
+	lopts.KernelDirect = parseAnnotationsBool(ctx, a, annotationKernelDirectBoot, lopts.KernelDirect)
+	if !lopts.KernelDirect {
+		lopts.KernelFile = uvm.KernelFile
+	}
+}
+
+// handleAnnotationPreferredRootFSType handles parsing annotationPreferredRootFSType and setting
+// implied annotations from the result
+func handleAnnotationPreferredRootFSType(ctx context.Context, a map[string]string, lopts *uvm.OptionsLCOW) {
+	lopts.PreferredRootFSType = parseAnnotationsPreferredRootFSType(ctx, a, annotationPreferredRootFSType, lopts.PreferredRootFSType)
+	switch lopts.PreferredRootFSType {
+	case uvm.PreferredRootFSTypeInitRd:
+		lopts.RootFSFile = uvm.InitrdFile
+	case uvm.PreferredRootFSTypeVHD:
+		lopts.RootFSFile = uvm.VhdFile
+	}
+}
+
+// handleAnnotationFullyPhysicallyBacked handles parsing annotationFullyPhysicallyBacked and setting
+// implied annotations from the result. For both LCOW and WCOW options.
+func handleAnnotationFullyPhysicallyBacked(ctx context.Context, a map[string]string, opts interface{}) {
+	switch opts.(type) {
+	case *uvm.OptionsLCOW:
+		lopts := (opts).(*uvm.OptionsLCOW)
+		lopts.FullyPhysicallyBacked = parseAnnotationsBool(ctx, a, annotationFullyPhysicallyBacked, lopts.FullyPhysicallyBacked)
+		if lopts.FullyPhysicallyBacked {
```

Comment:
Could set EnableDeferredCommit to false here also in case the user passed in true to avoid an unsupported config. Same would follow suit for WCOW below

Reply:
I think it EnableDeferredCommit is not explicitly needed for packing the UVM with PA, then it shouldn't be included here. If the user has EnableDeferredCommit to true here then pod creation will fail like normally.

Reply:
It's not it would just be a nicety. Do we actually know if pod creation fails with this setup? I just know it's unsupported I'm not sure on what occurs if you actually try and run it.

Reply:
With EnableDeferredCommit set to true? Didn't you add a check that fails it?

Reply:
Yes, it's in the working set PR. I'm saying I have not ran a container and seen what happens with this setup.

Reply:
Just double checked, pod creation fails in the HCS if EnableDeferredCommit is true when AllowOverCommit is false. I'm fine with this for now since we already have work in a PR to fail earlier.

Reply:
Ok cool :) Thats good to know